### DISCUSSION TITLE
How to get the name and uid of the user in %post section who started the build command?

### DISCUSSION THREAD

**Discussion by Heng-Zhou** [sylabs/singularity#2577] [Q&A] [answered]
I know we need to be root to build a Singularity container. But for some reason I need the name of the user and its uid in %post section who started the build command of the container. I tried `$(id -u)` and `$SUDO_USER`, but they did not work. Can you please teach me how I can bring the username and its uid to %post section, if I'm not going to hard-code it? The base OS is Ubuntu 22.04. Thank you.

**Comment by dtrudg** [comment, marked as answer]
There isn't an easy way to do this, because any environment related to the user, that might have been retained across `sudo`, is cleared when the `%post` is run.

You could use templating, and `--build-args` to pass the value in - see https://docs.sylabs.io/guides/latest/user-guide/definition_files.html#templating-how-to-pass-values-into-definition-files

Generally, using the username of the user who ran `sudo` should be avoided wherever possible in a container build, so that the container isn't inadvertently made specific to an individual user.

**Comment by Heng-Zhou** [comment]
Thank you.
